include/seqan3/alphabet/concept.hpp

```diff
@@ -43,219 +43,81 @@
-constexpr alphabet_type & assign_char(alphabet_type & c, underlying_char_t<alphabet_type> const in)
-    requires requires (alphabet_type c) { { c.assign_char(char{0}) } -> alphabet_type &; }
-{
-    return c.assign_char(in);
-}
-
-/*!\brief Free function wrapper that calls `.assign_char(in)` on the first argument.
- * \tparam alphabet_type Must provide an `.assign_char()` member function.
- * \param c An alphabet letter temporary.
- * \param in The `char` value you wish to assign.
- * \returns The assignment result as a temporary.
- * \details
- * Use this e.g. to newly create alphabet letters from char:
- * ~~~{.cpp}
- * auto l = assign_char(dna5{}, 'G');  // l is of type dna5
- * ~~~
+/*!\var rank_type seqan3::alphabet_size_v<semi_alphabet_concept>
+ * \brief The size of the alphabet. [value metafunction / constexpr variable template]
+ * \tparam alphabet_type Must satisfy seqan3::semi_alphabet_concept.
+ * \relates seqan3::semi_alphabet_concept
+ *
+ * This is the expression to retrieve the value (no parentheses!):
+ * ```cpp
+ * auto i = seqan3::alphabet_size_v<alphabet_type>; // will be of type seqan3::underlying_rank_t<alphabet_type>
+ * ```
  */
-template <typename alphabet_type>
-constexpr alphabet_type && assign_char(alphabet_type && c, underlying_char_t<alphabet_type> const in)
-    requires requires (alphabet_type c) { { c.assign_char(char{0}) } -> alphabet_type &; }
-{
-    return std::move(c.assign_char(in));
-}
-
-/*!\brief Free function wrapper that calls `.assign_rank(in)` on the first argument.
- * \tparam alphabet_type Must provide an `.assign_rank()` member function.
- * \param c The alphabet letter that you wish to assign to.
- * \param in The `rank` value you wish to assign.
- * \returns A reference to the alphabet letter you passed in.
+/*!\fn rank_type seqan3::to_rank(semi_alphabet_concept const a)
+ * \brief Returns the alphabet letter's value in rank representation.
+ * \relates seqan3::semi_alphabet_concept
+ * \param a The alphabet letter that you wish to convert to rank.
+ * \returns The letter's value in the alphabet's rank type (seqan3::underlying_rank).
  */
-template <typename alphabet_type>
-constexpr alphabet_type & assign_rank(alphabet_type & c, underlying_rank_t<alphabet_type> const in)
-    requires requires (alphabet_type c) { { c.assign_rank(uint8_t{0}) } -> alphabet_type &; }
-{
-    return c.assign_rank(in);
-}
-
-/*!\brief Free function wrapper that calls `.assign_rank(in)` on the first argument.
- * \tparam alphabet_type Must provide an `.assign_rank()` member function.
- * \param c An alphabet letter temporary.
- * \param in The `rank` value you wish to assign.
- * \returns The assignment result as a temporary.
- * \details
- * Use this e.g. to newly create alphabet letters from rank:
- * ~~~{.cpp}
- * auto l = assign_rank(dna5{}, 1);  // l is of type dna5 and == dna5::C
- * ~~~
+/*!\fn semi_alphabet_concept && seqan3::assign_rank(semi_alphabet_concept && a, rank_type const i)
```

Comment:
Again the one character names.. I would call them `letter` and `rank` (or `lttr`and `rnk`).